### PR TITLE
Implementing Canary Deployments on AWS ECS using AppMesh

### PR DESCRIPTION
**Pull request by srinivasreddych**
This blogPost helps us perform Canary Deployments for containers deployed on AWS ECS Fargate using AWS AppMesh.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: blogs/ecs-canary-deployments-pipeline/Makefile [hunk 1]
```diff
@@ -0,0 +1,30 @@
+set-env:
+	@chmod +x ./setup/scripts/export_environment_variables.sh
```
Comment (dastbe) [inline blogs/ecs-canary-deployments-pipeline/Makefile:2]: you can make these files executable ahead of time and git will honor that.

File: blogs/ecs-canary-deployments-pipeline/microservices/yelb-appserver/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,32 @@
+FROM public.ecr.aws/bitnami/ruby:2.4.10
```
Comment (dastbe) [inline blogs/ecs-canary-deployments-pipeline/microservices/yelb-appserver/Dockerfile:1]: q: do we not already have images in public ecr for yelb? do these assets diverge from yelb somehow?

**Review by dastbe — CHANGES_REQUESTED** [review]
Before I start digging deeper, I want to understand if its absolutely necessary to vendor yelb from source here. Can we use upstream images, or at least base our diffs on top of them?

File: blogs/ecs-canary-deployments-pipeline/microservices/yelb-appserver/Dockerfile (hunk 2 above)
Comment (srinivasreddych) [reply blogs/ecs-canary-deployments-pipeline/microservices/yelb-appserver/Dockerfile:1]: The yelb images have been sourced from Dockerhub and the official base images/yelb-images are not yet made available in public ECR. So we have done some testing with different versions of base images available in public ECR and started using the above declared.

File: blogs/ecs-canary-deployments-pipeline/Makefile (hunk 1 above)
Comment (srinivasreddych) [reply blogs/ecs-canary-deployments-pipeline/Makefile]: Made them executable ahead of time. Fixed it by removing references from Makefile

**Comment by tiagoReichert** [comment]
Hey @dastbe

We use the yelb sources as a sample app to demonstrate the pipeline (AWS CodeCommit and AWS CodePipeline are created for each yelb service). During the blog post we do changes to the source code that triggers the pipeline and builds the docker images to demonstrate the pipeline and the canary deployment (successful and rollback deployments).

**Comment by dastbe** [comment]
> We had to have a copy of the source code here instead of using directly from mreferre/yelb to prevent that any changes on that repo could break the steps we do on the blog post and because in our example we separated each service into it's own AWS CodeCommit repository to have a separate AWS CodePipeline for each service.

So the problem with vendoring yelb is that it brings in the licensing of yelb itself, for example I see in the vendored code

```
// Copyright (c) 2016 VMware, Inc. All Rights Reserved.
// This software is released under MIT license.
// The full license information can be found in LICENSE in the root directory of this project.
```

and no reference to the actual LICENSE file.

If we need to leverage yelb and modify its source as part of example, can we instead create a git submodule that links to the specific SHA of yelb? That way we aren't vendoring the code but we can still constrain ourselves to a specific version. If that's not possible, then we need to work with the owner of yelb to get it into something that is AWS owned with clear licensing.

**Comment by tiagoReichert** [comment]
Hey @dastbe, the Yelb source code was removed from this PR and we will reference a [forked repo](https://github.com/tiagoReichert/yelb) on the blog post for the sample app. Can you please approve it?

**Review by buzzsurfr — APPROVED** [review]
LGTM :shipit:

**Comment by srinivasreddych** [comment]
Hello @dastbe . Can you please take a look?

**Pull request merged**